src/lib/Microsoft.Health.Extensions.Fhir.R4/FhirClientValidator.cs

```diff
@@ -4,22 +4,26 @@
 // -------------------------------------------------------------------------------------------------
 
 using System;
+using System.Threading.Tasks;
 using EnsureThat;
-using Hl7.Fhir.Rest;
 using Microsoft.Health.Extensions.Fhir.Telemetry.Exceptions;
+using Microsoft.Health.Fhir.Client;
 using Microsoft.Health.Logging.Telemetry;
 
 namespace Microsoft.Health.Extensions.Fhir
 {
-    public static class FhirServiceValidator
+    public static class FhirClientValidator
     {
-        public static bool ValidateFhirService(FhirClient client, ITelemetryLogger logger)
+        public static async Task<bool> ValidateFhirClientAsync(
+            this IFhirClient client,
+            ITelemetryLogger logger)
         {
             EnsureArg.IsNotNull(client, nameof(client));
+            EnsureArg.IsNotNull(logger, nameof(logger));
 
             try
             {
-                client.CapabilityStatement(SummaryType.True);
+                await client.ReadAsync<Hl7.Fhir.Model.CapabilityStatement>("metadata").ConfigureAwait(false);
```

Comment:
>CapabilityStatement [](http://example.com/codeflow?start=54&length=19)

Does this preserve the summary functionality from the old call?  I don't think so.  Here is the source from the FirelyClient. Would be good if we could preserve [https://github.com/FirelyTeam/firely-net-sdk/blob/471a5a1417355e4bbcfbff722b62d624c0f946e1/src/Hl7.Fhir.Core/Rest/TransactionBuilder.cs#L209](https://github.com/FirelyTeam/firely-net-sdk/blob/471a5a1417355e4bbcfbff722b62d624c0f946e1/src/Hl7.Fhir.Core/Rest/TransactionBuilder.cs#L209)

Capability statements can be quite large and we are just doing this call to verify we can connect to the FHIR service (metadata end point doesn't need auth) so a summary is fine.

Reply:
Thanks for calling this out. I've gone ahead and updated the GET call to return all supported elements that are marked as "summary" in the base definition of the resource